### PR TITLE
Add NrrdWriter for multi-channel arrays

### PR DESCRIPTION
**Pull request by markus-hinsche**
## Motivation

The `Writer` class (in `monailabel/transform/writer.py`) can't handle multi-channel arrays. The ITK reader and writer both can't handle multi-channel arrays, therefore we want to introduce a new writer that can do this.

**Related PRs**
This work is related to other works in this area:
* After the NRRD reader, writer also makes sense, see issues:
    * see issue https://github.com/Project-MONAI/MONAI/issues/4238 with PR https://github.com/Project-MONAI/MONAI/pull/4259
* `seg.nrrd` tooling with 3D Slicer
    * https://github.com/Project-MONAI/MONAILabel/pull/768

## Changes

* Add new function `write_seg_nrrd()` (similar to `write_itk()`)
* Add condition when to use nrrd writer
* Add `pynrrd` pip dependency to project
* Add unit test for writing multi-channel data

### CODE REVIEW

**Comment by SachidanandAlle** [comment]
Thanks for raising the PR.. will be look into this asap..

File: monailabel/transform/writer.py [hunk 1]
```diff
@@ -58,6 +60,85 @@ def write_itk(image_np, output_file, affine, dtype, compress):
     itk.imwrite(result_image, output_file, compress)
 
 
+def write_seg_nrrd(
+    image_np: np.ndarray,
+    output_file: str,
+    dtype: type,
```
Comment (SachidanandAlle) [inline monailabel/transform/writer.py:66]: may be u can have default values for dtype

File: monailabel/transform/writer.py [hunk 2]
```diff
@@ -104,15 +185,28 @@ def __call__(self, data):
             output_file = tempfile.NamedTemporaryFile(suffix=ext).name
             logger.debug(f"Saving Image to: {output_file}")
 
+            if self.is_multichannel_image(image_np):
+                if ext != ".seg.nrrd":
+                    logger.debug(
```
Comment (SachidanandAlle) [inline monailabel/transform/writer.py:190]: this can be logger.warning

File: monailabel/transform/writer.py [hunk 3]
```diff
@@ -104,15 +185,28 @@ def __call__(self, data):
...
+            elif self.nibabel and ext.lower() in [".nii", ".nii.gz"]:
                 logger.debug("Using MONAI write_nifti...")
                 write_nifti(image_np, output_file, affine=affine, output_dtype=dtype)
             else:
                 write_itk(image_np, output_file, affine, dtype, compress)
 
         return output_file, output_json
 
+    def is_multichannel_image(self, image_np):
+        return len(image_np.shape) == 4 and image_np.shape[-1] > 1
```
Comment (SachidanandAlle) [inline monailabel/transform/writer.py:208]: channel last?  normally all images/labels/predictions are processed as channel first in post transform... we better cross check by running some examples e2e.. 

@diazandr3s  can help to verify this feature...

Comment (diazandr3s) [reply monailabel/transform/writer.py:208]: Happy to help :)
Should we wait until this PR (https://github.com/Project-MONAI/MONAI/pull/4259) is merged?

Comment (SachidanandAlle) [reply monailabel/transform/writer.py:208]: you can checkout the remote branch and verify..

File: monailabel/transform/writer.py (hunk 1 above)
Comment (markus-hinsche) [reply monailabel/transform/writer.py:66]: We tried to make this function's signature similar to the `write_itk()`. It is probably best to also have a default there as well then, would you agree?

File: monailabel/transform/writer.py (hunk 3 above)
Comment (markus-hinsche) [reply monailabel/transform/writer.py:208]: https://github.com/Project-MONAI/MONAI/pull/4259 is merged now. I merged upstream/main into this branch

Comment (markus-hinsche) [reply monailabel/transform/writer.py]: Indeed this seemed wrong, I updated the code to check the first dimension instead.

FYI:

**[pynrrd](https://pypi.org/project/pynrrd/)** choses the following convention
```
default index_order='F'
nrrd.writer.write(data)
index_order : {'C', 'F'}, optional
    Specifies the index order used for writing. Either 
    'C' (C-order) 
    	where the dimensions are ordered from
    	slowest-varying to fastest-varying (e.g. (z, y, x)), 
    'F' (Fortran-order)
    	where the dimensions are ordered
    	from fastest-varying to slowest-varying (e.g. (x, y, z)).
```
We chose 'C' as a default here

**torch** has the following convention: `NxCxHxW`

At the moment, in our monailabel-app, we use (5, 4280, 3520, 1) = (channels, width, height, batch)

**Comment by SachidanandAlle** [comment]
can you please fix the base branch.. it complains out-of-date and can't be done through UI

**Comment by markus-hinsche** [comment]
> can you please fix the base branch.. it complains out-of-date and can't be done through UI

I am not sure what the problem is. Feel free to elaborate

**Review by diazandr3s — APPROVED** [review]
Thanks for the PR. Looks good to me.

**Comment by markus-hinsche** [comment]
Great! Thanks for the feedback

**Review by SachidanandAlle — APPROVED** [review]

**Comment by SachidanandAlle** [comment]
Can't merge as it says out of date

**Comment by SachidanandAlle** [comment]
I see the problem.. u have forked the forked repo..
https://github.com/jlvahldiek/MONAILabel => yours..
this is bit confusing for github itself..

can you please fix the fork and resolve the conflicts correct.. if possible i suggest please fork from the original repo..

**Comment by markus-hinsche** [comment]
Closing in favor of https://github.com/Project-MONAI/MONAILabel/pull/793

**Pull request closed without merging**